File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        memo = {}
        def dp(idx):
            if idx > n:
                return 0
            if idx == n:
                return 1
            if idx in memo:
                return memo[idx]
            memo[idx] = dp(idx+1) + dp(idx+2)
            return memo[idx]
        return dp(0)

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let memo = new Map()
    var dp = (idx)=>{
        if (idx === n){
            return 1;
        }
        if (idx > n){
            return 0;
        }
        if (memo.has(idx)){
            return memo.get(idx);
        }
        let tot = 0
        tot += dp(idx+1)
        tot += dp(idx+2)
        memo.set(idx,tot)
        return tot
    }
    return dp(0)
    
};